### PR TITLE
onBuildWidgetFailed - handle build errors

### PR DESCRIPTION
**Pull request by Thauanny**
## Description

Implemented option to handle errors when building a component. So we can have better control over what to do in case of failure.

### CODE REVIEW

File: json_dynamic_widget/lib/src/components/json_widget_registry.dart [hunk 1]
```diff
@@ -80,7 +81,7 @@ class JsonWidgetRegistry {
   final JsonWidgetRegistry? _parent;
   final _values = <String?, dynamic>{};
   late List<ArgProcessor> _argProcessors;
-
+  final Widget Function(Object error, BuildContext context)? onBuildWidgetFailed;
```
Comment (crisperit) [inline json_dynamic_widget/lib/src/components/json_widget_registry.dart:84]: Woth to pass also the Stacktrace and let's type error as `dynamic` just to be consistent

Comment (Thauanny) [reply json_dynamic_widget/lib/src/components/json_widget_registry.dart:84]: i can try it

Comment (Thauanny) [reply json_dynamic_widget/lib/src/components/json_widget_registry.dart]: pls check it out the update!

**Review by crisperit — APPROVED** [review]
lgtm

**Comment by jpeiffer** [comment]
Wow, this is great!  Thanks so much for the contribution!

**Comment by jpeiffer** [comment]
I'll get it deployed later this week with the next release.

**Pull request merged**